jscomp/others/belt_Map.mli

```diff
@@ -12,53 +12,23 @@
 (*  Adapted by authors of ReScript without using functors          *)
 (***********************************************************************)
 
-(** A _immutable_ sorted map module which allows customize _compare_ behavior.
+(** The top level provides generic immutable map operations.
 
-    The implementation uses balanced binary trees, and therefore searching
-    and insertion take time logarithmic in the size of the map.
-
-    For more info on this module's usage of identity, `make` and others, please see
-    the top level documentation of Belt, **A special encoding for collection safety**.
-
-    Example usage:
-
-    ```
-    module PairComparator = Belt.Id.MakeComparable(struct
-        type t = int * int
-        let cmp (a0, a1) (b0, b1) =
-          match Pervasives.compare a0 b0 with
-          | 0 -> Pervasives.compare a1 b1
-          | c -> c
-      end)
-
-    let myMap = Belt.Map.make ~id:(module PairComparator)
-    let myMap2 = Belt.Map.set myMap (1, 2) "myValue"
-    ```
-
-    The API documentation below will assume a predeclared comparator module for integers, IntCmp
+    It also has three specialized inner modules `Belt.Map.Int`, `Belt.Map.String` and `Belt.Map.Dict`.
 *)
 
 
-(** Specalized when key type is `int`, more efficient
-    than the generic type, its compare behavior is fixed using the built-in comparison
-*)
 module Int = Belt_MapInt
 
-(** specalized when key type is `string`, more efficient
-    than the generic type, its compare behavior is fixed using the built-in comparison *)
 module String = Belt_MapString
 
-(** This module seprate identity from data, it is a bit more verboe but slightly
-    more efficient due to the fact that there is no need to pack identity and data back
-    after each operation
-
-    **Advanced usage only**
-*)
 module Dict = Belt_MapDict
 
 
 type ('key, 'value, 'identity) t
-(** `('key, 'identity) t`
+(** ```res prelude
```

Comment:
I didn't mention `res prelude` snippets... those should be kinda "invisible" to the reader of the docs. Could we move those code fences in a simple `(* *)` comment to preserve them?